thumbor/compatibility/compat.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# thumbor imaging service
+# https://github.com/thumbor/thumbor/wiki
+
+# Licensed under the MIT license:
+# http://www.opensource.org/licenses/mit-license
+# Copyright (c) 2011 globo.com thumbor@googlegroups.com
+
+import asyncio
+
+
+async def compatibility_get(*args, **kw):
+    func = kw.pop("func", None)
+    if func is None:
+        raise RuntimeError(
+            "'func' argument can't be None when calling thumbor's compatibility layer."
+        )
+
+    loop = asyncio.get_event_loop()
+    queue = asyncio.Queue()
+
+    def put(*args):
+        loop.call_soon_threadsafe(queue.put_nowait, args)
+
+    func(*args, **kw, callback=put)
```

Comment:
There is a good chance I don't understand what's going on...
But this call here is synchronous and blocking. The fact that you put it's  result into queue does not solve anything. Consider the following example where my func is syncronous:
```python
def func(callback):
    sleep(10)
    callback('Great success')
```
in my understanding, such `func` would block the loop until it is finished sleeping. Then it'll call callback, which would put `Great success` string into a queue and then on next loop iteration it'll be popped from queue. However, that does not matter, since plain call to `func` is blocking right here.

Reply:
Hey, thanks for the review! I think maybe we are mixing two different concepts here: non-blocking and async (as in AsyncIO async).

The compatibility layer is not trying to turn blocking storages, loaders and result storages into non-blocking ones. If the storage is blocking it will remain blocking.

If the storage uses non-blocking code with callbacks, it should remain non-blocking. If it is not the case, it should remain blocking. Maybe picking the legacy file loader was not the best scenario as it is blocking. I think I'll change to use the http loader instead as it has non-blocking callback based code.

Reply:
Hmm just tried. Using the legacy http loader is not possible as the callback based fetch has been removed from tornado's http client.

Just to clarify further, this change is to enable loaders, storages and result storages that were working with thumbor 6.x to keep working with thumbor 7 with as small a change as possible (no need to use AsyncIO in any calls).

I'm not sure on how feasible that is given that Python 3 has different requirements on code and modern Tornado has changed quite a bit. It Anyway I believe it was worth the shot. If we can test with tc_aws and it makes it easier it would be great.